DeezyMatch/tests/test_pipeline_one_col_input.py

```diff
@@ -86,4 +86,39 @@ def test_pipeline_one_col_input():
                          pretrained_model_path="./models/finetuned_test003/finetuned_test003.model",
                          pretrained_vocab_path="./models/finetuned_test003/finetuned_test003.vocab",
                          number_test_rows=5)
-    
\ No newline at end of file
+    
+    from DeezyMatch import candidate_ranker
+    # Select candidates based on L2-norm distance (aka faiss distance):
+    # find candidates from candidate_scenario
+    # for queries specified in query_scenario
+    candidates_pd_predtrue = \
+        candidate_ranker(query_scenario="./combined_003/queries_test",
+                         candidate_scenario="./combined_003/candidates_test",
+                         ranking_metric="faiss",
+                         selection_threshold=5.,
+                         num_candidates=2,
+                         search_size=10,
+                         use_predict=False,
+                         output_path="ranker_results_003/test_candidates_deezymatch",
+                         pretrained_model_path="./models/finetuned_test003/finetuned_test003.model",
+                         pretrained_vocab_path="./models/finetuned_test003/finetuned_test003.vocab",
+                         number_test_rows=5)
+    
+    from DeezyMatch import candidate_ranker
+    # Select candidates based on L2-norm distance (aka faiss distance)
+    # where ranking_metric is conf and use_prediction is false:
+    candidates_pd_predfalse = \
+        candidate_ranker(query_scenario="./combined_003/queries_test",
+                         candidate_scenario="./combined_003/candidates_test",
+                         ranking_metric="faiss",
+                         selection_threshold=5.,
+                         num_candidates=2,
+                         search_size=10,
+                         use_predict=True,
+                         output_path="ranker_results_003/test_candidates_deezymatch",
+                         pretrained_model_path="./models/finetuned_test003/finetuned_test003.model",
+                         pretrained_vocab_path="./models/finetuned_test003/finetuned_test003.vocab",
+                         number_test_rows=5)
+    
+    # Same candidates and faiss scores should be retrieved independently of use_predict value:
+    candidates_pd_predtrue.faiss_distance == candidates_pd_predfalse.faiss_distance
```

Comment:
Can we also add a test for `length_diff` please?

Reply:
Hi @kasra-hosseini, good idea. Done [here](https://github.com/Living-with-machines/DeezyMatch/pull/121/commits/daa3ee6d214bcabd081d0e3cedb42b7387a78955#diff-55b2da504d620c8846599dd56717ccd4ee90f38cc18db8daa0043cc27fc267e9).

Reply:
Thanks